### PR TITLE
Add snapshot tests

### PR DESCRIPTION
**Pull request by brekk**
## Summary

Add snapshot testing via `jest`.

## Testing Plan

- `nps test` - run tests
- `nps test.snapshot` - update snapshots
- `nps test.watch` - run tests on locally changed files

## Breaking Change

Nein.

### CODE REVIEW

File: __tests__/page-snapshot-user-settings-not-authorized.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        query: { id: '111' },
+      },
+    })
+    await waitForElementToBeRemoved(() =>
+      screen.queryByRole('alert', { name: 'Loading' })
+    )
+    expect(container).toMatchSnapshot()
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.warn('cool', e)
```
Comment (leanthebean) [inline __tests__/page-snapshot-user-settings-not-authorized.tsx:33]: if this test fails it'll console log out the message "cool"? if so, can we make it more descriptive?

File: __tests__/page-snapshot-user-settings.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import User from 'pages/users/[id]'
+import { screen, waitForElementToBeRemoved } from '@testing-library/react'
+import { render } from 'jest.utils'
+import fetch from 'jest-fetch-mock'
+import { mockUserPage } from './page-snapshot-user'
+
+beforeEach(() => {
+  fetch.resetMocks()
```
Comment (leanthebean) [inline __tests__/page-snapshot-user-settings.tsx:8]: why do you need to reset mocks before every test? there's only one test

File: __tests__/page-snapshot-user-settings.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        query: { id: '111' },
+      },
+    })
+    await waitForElementToBeRemoved(() =>
+      screen.queryByRole('alert', { name: 'Loading' })
+    )
+    expect(container).toMatchSnapshot()
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.warn('cool', e)
```
Comment (leanthebean) [inline __tests__/page-snapshot-user-settings.tsx:28]: again, "cool" is not descriptive. we shouldn't be fighting with our linter. looks like you're just doing this to use Error`e` so the linter doesn't yell at you for unused variables? maybe fix that issue instead?

File: __tests__/page-snapshot-user.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    : matches('/events?')
+    ? FIXTURE.events
+    : endsWith('granularity=lifetime')
+    ? FIXTURE.allTimeMetrics
+    : matches('/metrics?granularity=total')
+    ? FIXTURE.weeklyMetrics
+    : matches('/metrics/config')
+    ? FIXTURE.metricsConfig
+    : { error: "You haven't implemented this route yet" }
+  // console.log({ url: req.url, body, init: goodResponse })
```
Comment (leanthebean) [inline __tests__/page-snapshot-user.tsx:112]: remove unused console log comment

File: components/Loader/index.tsx [hunk 5]
```diff
@@ -17,7 +17,12 @@ export const Loader = () => (
         </clipPath>
       </defs>
     </svg>
-    <div className={styles.fishScale}>
+    <div
```
Comment (leanthebean) [inline components/Loader/index.tsx:20]: Looks like this doesn't belong in this PR? 

Let's keep our PRs small and manageable. If we already have a 13k PR that introduces a different paradigm for testing, let's not throw in other changes as well. We can make a separate PR for this.

File: cypress/integration/pages/about.ts [hunk 6]
```diff
@@ -47,21 +47,11 @@ describe('/about', () => {
         text: 'Submit an issue ',
         href: 'https://github.com/iron-fish/ironfish/issues',
       },
-      {
```
Comment (leanthebean) [inline cypress/integration/pages/about.ts:50]: what's happening here? why was this removed? it sounds like this PR's goal is to introduce snapshot testing, and this change makes it look like other changes were made.

File: __tests__/component-select.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  const { container } = selectWithProps({
+    disabled: false,
+    options,
+    defaultValue: 'a',
+    value: 'a',
+    onChange: jest.fn(),
+    setTouched: jest.fn(),
+    className: 'test',
+  })
+  expect(container).toMatchSnapshot()
```
Comment (leanthebean) [inline __tests__/component-select.tsx:21]: im confused what this is testing. the values (options) are exactly what is in the snapshot. it's just testing that the snapshot that we ourselves generate has the correct values?

**Review by leanthebean — CHANGES_REQUESTED** [review]
first pass

File: __tests__/page-snapshot-user-settings.tsx (hunk 2 above)
Comment (brekk) [reply __tests__/page-snapshot-user-settings.tsx:8]: Incorrect, the mocks are persisted across runs.

File: components/Loader/index.tsx (hunk 5 above)
Comment (brekk) [reply components/Loader/index.tsx:20]: It belongs in this PR so that we can test against the `role`.
